bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/item/ItemResource.java

```diff
@@ -385,7 +390,7 @@ public Response removeMember(@PathParam("itemName") @ApiParam(value = "item name
             @ApiResponse(code = 404, message = "Item not found or item is not editable.") })
     public Response removeItem(@PathParam("itemname") @ApiParam(value = "item name", required = true) String itemname) {
 
-        if (managedItemProvider.remove(itemname) == null) {
+        if (itemname == null || managedItemProvider.remove(itemname) == null) {
```

Comment:
Hm, shouldn't we respond with a `bad request` if the item name is not given?
The parameter is marked "required" and if it is null, it is missing on the request.

```Java
if (itemname == null) {
    return Response.status(Status.BAD_REQUEST).build();
}
```

Reply:
Sure, I did not think about this, I just wanted to fix the possible NPE which the IDE pointed me to :)